### PR TITLE
Tinygrad Inference Router

### PR DESCRIPTION
**Pull request by reddyn12**
Add logic to be able to support building different model types in tinygrad inference

### CODE REVIEW

**Comment by reddyn12** [comment]
I can move the ```build_llama``` into ```llama.py``` and rename the function to ```build_model```. This will make the ```if/else and raise``` not necessary. lmk how I should approach this

File: exo/inference/tinygrad/inference.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
 }
 
 
-def build_transformer(model_path: Path, shard: Shard, model_size="8B", device=None):
+def build_llama(model_path: Path, shard: Shard, model_size="8B", device=None):
```
Comment (dtnewman) [inline exo/inference/tinygrad/inference.py:30]: why change this to `build_llama`?

Comment (reddyn12) [reply exo/inference/tinygrad/inference.py:30]: Every model will have its own build function. So build_llava, build_llama, build_mamba, etc...

Comment (dtnewman) [reply exo/inference/tinygrad/inference.py:30]: ah, i missed that you changed it below. These AsyncIO calls are not quite as reader friendly 😄

**Comment by reddyn12** [comment]
@AlexCheema is this proposal fine, or is there a better way to do this?

**Comment by AlexCheema** [comment]
Ideally we do this like #139 where we support models automatically without needing separate `build_{model}` functions.

**Comment by reddyn12** [comment]
Ok, I'll close this pr, do the LLaVa tinygrad implementation pr with this kind of logic, and then put up another pr that refactors the inference logic to used the stateful_shards.

**Pull request closed without merging**